packages/editor/src/components/rich-text/index.js

```diff
@@ -497,15 +497,7 @@ export class RichText extends Component {
 			return;
 		}
 
-		const empty = this.isEmpty();
-
-		// It is important to consider emptiness because an empty container
-		// will include a padding BR node _after_ the caret, so in a forward
-		// deletion the isHorizontalEdge function will incorrectly interpret the
-		// presence of the BR node as not being at the edge.
-		const isEdge = ( empty || isHorizontalEdge( this.editableRef, isReverse ) );
```

Comment:
This is no longer needed with the changes to `isHorizontalEdge`. It's now generalised. `isHorizontalEdge` should return true for the visual edge (you can never reach empty text nodes or trailing BR elements), not for the true DOM edge.